src/data_info/services/path_lists.clj

```diff
@@ -135,30 +138,56 @@
     (doseq [path paths]
       (validators/not-base-path user path))))
 
-(defn create-path-list
+(defn create-path-list*
   "Creates an HT Path List file from the given set of paths and filtering params.
    The resulting HT Path List file will contain only file or only folder paths (depending on the `folders-only` param),
    but no paths of folders in the initial given list will be included.
    If `recursive` is true, then all subfolders (plus all their files and subfolders)
    of any given folder paths are parsed and filtered as well.
    Throws an error if the filtering params result in no matching paths."
+  [irods {:keys [user dest path-list-info-type name-pattern info-type folders-only recursive]
+    :or   {path-list-info-type (cfg/ht-path-list-info-type)}}
+   {:keys [paths]}]
+  (otel/with-span [s ["create-path-list"]]
+    (validate-request-paths irods user dest paths)
+
+    (let [path-list-contents  (paths->path-list irods
+                                                user
+                                                (info-type->file-identifier path-list-info-type)
+                                                name-pattern
+                                                info-type
+                                                folders-only
+                                                recursive
+                                                paths)
+          path-list-file-stat (with-in-str path-list-contents (copy-stream @(:jargon irods) *in* user dest))]
+      (irods/with-jargon-exceptions [admin-cm]
+        (filetypes/add-type-to-validated-path admin-cm dest path-list-info-type))
+      (rods/invalidate irods dest)
+      {:file (stat/decorate-stat irods user (cfg/irods-zone) path-list-file-stat (process-filters nil nil))})))
+
+(defn- create-path-list-thread
+  [async-task-id]
+  (let [irods-fn (fn [irods async-task update-fn]
+                   (let [{:keys [username data]} async-task]
+                     (create-path-list* irods data (:paths data))))
```

Comment:
It looks like `create-path-list*` still wants to destructure `paths` from a map in its 3rd argument, but now `paths` are being passed in as a raw list in the 3rd arg here?

Reply:
Oh, good catch for sure, not sure why my eyes skipped over that.